src/Multicall3.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.12;
+
+/// @title Multicall3
+/// @notice Aggregate results from multiple read-only function calls
+/// @dev Multicall & Multicall2 backwards-compatible
+/// @author Michael Elliot <mike@makerdao.com>
+/// @author Joshua Levine <joshua@makerdao.com>
+/// @author Nick Johnson <arachnid@notdot.net>
+/// @author Andreas Bigger <andreas@nascent.xyz>
+
+contract Multicall3 {
+    struct Call {
+        address target;
+        bytes callData;
+    }
+
+    struct Call3 {
+        address target;
+        bool allowFailure;
+        bytes callData;
+    }
+
+    struct Result {
+        bool success;
+        bytes returnData;
+    }
+
+    function aggregate(Call[] calldata calls) public returns (uint256 blockNumber, bytes[] memory returnData) {
```

Comment:
I love properly commented code so what about some natspec comments here (at least `param`). Applies to all functions generally:
```solidity
    /// @notice ...
    /// @dev function code is super gas optimised ;-) 
    /// @param ...
    /// @return ...
```

Reply:
I also love properly commented code, though IMO this code is super straight forward to read so I don't think comments add much value here, and afaik no one has issues with usage/lack of comments in the prior versions 🙂